build_runner/lib/src/build_script_generate/build_script_generate.dart

```diff
@@ -15,30 +15,135 @@ const scriptLocation = '.dart_tool/build/build.dart';
 Future<Null> ensureBuildScript() async {
   var scriptFile = new io.File(scriptLocation);
   // TODO - how can we invalidate this?
-  if (scriptFile.existsSync()) return;
+  //if (scriptFile.existsSync()) return;
   scriptFile.createSync(recursive: true);
   await scriptFile.writeAsString(await _generateBuildScript());
 }
 
+final _packageGraph = new TypeReference((b) => b
+  ..symbol = 'PackageGraph'
+  ..url = 'package:build_runner/build_runner.dart');
+
+final _buildAction = new TypeReference((b) => b
+  ..symbol = 'BuildAction'
+  ..url = 'package:build_runner/build_runner.dart');
+
+final _buildActions = new TypeReference((b) => b
+  ..symbol = 'List'
+  ..types.add(_buildAction));
+
 Future<String> _generateBuildScript() async {
   var packageGraph = new PackageGraph.forThisPackage();
   var buildConfigs =
-      await Future.wait(packageGraph.orderedPackages.map(_packageBuildConfig));
-  var printStatements = buildConfigs
-      .where((config) => config.builderDefinitions.isNotEmpty)
-      .map((config) => refer('print').call([
-            literalString('I would run: \'${config.packageName} - '
-                '${config.builderDefinitions.keys.toList()}')
-          ]).statement);
-  final library = new File((b) => b.body.addAll([
-        new Method.returnsVoid((b) => b
-          ..name = 'main'
-          ..body = new Block((b) => b..statements.addAll(printStatements)))
-      ]));
-  final emitter = new DartEmitter();
+      (await Future.wait(packageGraph.orderedPackages.map(_packageBuildConfig)))
+          .where((c) => c.builderDefinitions.isNotEmpty);
+  final library = new File(
+      (b) => b.body.addAll([_findBuildActions(buildConfigs), _main()]));
+  final emitter = new DartEmitter(new Allocator.simplePrefixing());
   return new DartFormatter().format('${library.accept(emitter)}');
 }
 
+Method _findBuildActions(Iterable<BuildConfig> buildConfigs) =>
+    new Method((b) => b
+      ..name = '_buildActions'
+      ..requiredParameters.add(new Parameter((b) => b
+        ..name = 'graph'
+        ..type = _packageGraph))
+      ..returns = _buildActions
+      ..body = new Block((b) => b
+        ..statements.addAll([
+          _buildActions.newInstance([]).assignVar('actions').statement,
+          // TODO - Pass actual arguments
+          literalList([], new TypeReference((b) => b..symbol = 'String'))
+              .assignVar('args')
+              .statement,
+        ]
+          ..addAll(_addBuildActions(buildConfigs))
```

Comment:
it is a bit hard to tell what this corresponds to, I would consider using regular function syntax and removing some of the cascades.

Reply:
Done